src/core/components/scroll-to-top.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react"
+
+const ScrollToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [hideTimeout, setHideTimeout] = useState(null) 
+
+  const styles = {
```

Comment:
Inline styles are not used in this project, so this code doesn't follow the convention.

Reply:
And for the future. Since the styles object doesn’t depend on any props, it’s best to define it outside the component. This avoids redefining the styles object on each render, improving performance.